File: android/ModuMessenger/app/src/main/java/com/example/modumessenger/dto/ChatDto.java
package com.example.modumessenger.dto;

public class ChatDto {
    private  Long id;
    private int chatType;
    private String roomId;
    private String sender;
    private String message;
    private String chatTime;

    public Long getId() { return this.id; }
    public String getRoomId() { return this.roomId; }
    public String getSender() { return this.sender; }
    public String getMessage() { return this.message; }
    public String getChatTime() { return this.chatTime; }
    public int getChatType() { return this.chatType; }

    public void setId(Long id) { this.id = id; }
    public void setRoomId(String roomId) { this.roomId = roomId; }
    public void setSender(String sender) { this.sender = sender; }
    public void setMessage(String message) { this.message = message; }
    public void setChatTime(String chatTime) { this.chatTime = chatTime; }
    public void setChatType(int chatType) { this.chatType = chatType; }
}

File: android/modu-chat/app/src/main/kotlin/com/example/modumessenger/data/dto/ChatDto.kt
package com.example.modumessenger.data.dto

import com.google.gson.annotations.SerializedName

/** 소켓으로도 이 모양 그대로 오간다(보낼 때 `id` 는 null, 서버가 붙인다). */
data class ChatDto(
    val id: Long? = null,
    val chatType: Int = 0,
    val roomId: String? = null,
    val sender: String? = null,
    val message: String? = null,
    val chatTime: String? = null,
    val reactions: List<ReactionSummaryDto>? = null,
)

/** 서버의 ReactionSummaryDto 와 같은 모양. 이력 조회와 REACTION 프레임 둘 다 이걸 쓴다. */
data class ReactionSummaryDto(
    val emoji: String? = null,
    val count: Int = 0,
    val userIds: List<String>? = null,
)

data class ChatRoomDto(
    val roomId: String? = null,
    val roomName: String? = null,
    val roomImage: String? = null,
    val lastChatMsg: String? = null,
    val lastChatId: String? = null,
    val lastChatTime: String? = null,
    val members: List<MemberDto>? = null,
)

data class ChatRoomUnreadDto(
    @SerializedName("roomId") val roomId: String? = null,
    @SerializedName("lastSendChatId") val lastSendChatId: Long = 0L,
    @SerializedName("lastReadChatId") val lastReadChatId: Long = 0L,
    @SerializedName("unreadChatCount") val unreadChatCount: Long = 0L,
)

data class ChatReadCursorDto(
    @SerializedName("userId") val userId: String? = null,
    @SerializedName("lastReadChatId") val lastReadChatId: Long = 0L,
)
